multivariate_grouped_data_plot.Rmd

```diff
@@ -0,0 +1,825 @@
+- 如果只想添加较高的误差线，而不是较低的误差线，则使用ymin = len（而不是len-sd）和ymax = len + sd。
+
+```{r}
+# Initialize ggplot with data
+f <- ggplot(
+  df.summary, 
+  aes(x = dose, y = len, ymin = len-sd, ymax = len+sd)
+  )
+```
+
+(4) 创建简单的误差图
+
+```{r}
+# Vertical line with point in the middle
+f + geom_pointrange()
+# Standard error bars
+f + geom_errorbar(width = 0.2) +
+  geom_point(size = 1.5)
+```
+
+(5) 创建水平误差图。 将dose放在y轴上，len放在x轴上。指定xmin和xmax。
+
+```{r}
+# Horizontal error bars with mean points
+# Change the color by groups
+ggplot(
+  df.summary, 
+  aes(x = len, y = dose, xmin = len-sd, xmax = len+sd)
+  ) +
+  geom_point(aes(color = dose)) +
+  geom_errorbarh(aes(color = dose), height=.2)+
+  theme_light()
+```
+
+(6) 添加抖动点（代表单个点），点图和小提琴图。
+
+为此，应使用原始数据（df）初始化ggplot并在误差绘图功能（此处为geom_pointrange()）中指定df.summary数据。
+
+```{r}
+# Combine with jitter points
+ggplot(df, aes(dose, len)) +
+  geom_jitter(
+    position = position_jitter(0.2), color = "darkgray"
+    ) + 
+  geom_pointrange(
+    aes(ymin = len-sd, ymax = len+sd),
+    data = df.summary
+    )
+# Combine with violin plots
+ggplot(df, aes(dose, len)) +
+  geom_violin(color = "darkgray", trim = FALSE) + 
+  geom_pointrange(
+    aes(ymin = len-sd, ymax = len+sd),
+    data = df.summary
+    )
+```
+
+(7) 创建平均+/-误差的基本条形图/线图。 
+
+我们只需要df.summary数据。
+
+为线图添加上下误差线：ymin = len-sd和ymax = len + sd。
+
+为条形图仅添加较高的误差线：ymin = len（而不是len-sd）和ymax = len + sd。
+
+```{r}
+# (1) Line plot
+ggplot(df.summary, aes(dose, len)) +
+  geom_line(aes(group = 1)) +
+  geom_errorbar( aes(ymin = len-sd, ymax = len+sd),width = 0.2) +
+  geom_point(size = 2)
+# (2) Bar plot
+ggplot(df.summary, aes(dose, len)) +
+  geom_bar(stat = "identity", fill = "lightgray", 
+           color = "black") +
+  geom_errorbar(aes(ymin = len, ymax = len+sd), width = 0.2) 
+```
+
+对于线图，我们也可以将x轴视为数字：
+
+```{r}
+df.sum2 <- df.summary
+df.sum2$dose <- as.numeric(df.sum2$dose)
+ggplot(df.sum2, aes(dose, len)) +
+  geom_line() +
+  geom_errorbar( aes(ymin = len-sd, ymax = len+sd),width = 0.2) +
+  geom_point(size = 2)
+```
+
+(8) 条形图和折线图+抖动点。 
+
+我们需要用于抖动点的原始df数据和其他geom层的df.summary数据。
+
+对于线图：首先，添加抖动点，然后在抖动点顶部添加线条+误差线+均值点。
+
+对于条形图：首先，添加条形图，然后在条形图的顶部添加抖动点+误差线。
+
+```{r}
+# (1) Create a line plot of means + 
+# individual jitter points + error bars 
+ggplot(df, aes(dose, len)) +
+  geom_jitter( position = position_jitter(0.2),
+               color = "darkgray") + 
+  geom_line(aes(group = 1), data = df.summary) +
+  geom_errorbar(
+    aes(ymin = len-sd, ymax = len+sd),
+    data = df.summary, width = 0.2) +
+  geom_point(data = df.summary, size = 2)
+# (2) Bar plots of means + individual jitter points + errors
+ggplot(df, aes(dose, len)) +
+  geom_bar(stat = "identity", data = df.summary,
+           fill = NA, color = "black") +
+  geom_jitter( position = position_jitter(0.2),
+               color = "black") + 
+  geom_errorbar(
+    aes(ymin = len-sd, ymax = len+sd),
+    data = df.summary, width = 0.2) 
+```
+
+#### 3.7.2 多个组的均值/中位数图。 
```

Comment:
Hashtag headers should not be followed by numbers since the hashtags will create numbered subheadings.